### PR TITLE
Correct documented return values for certain ENV methods

### PR DESCRIPTION
**Pull request by BurdetteLamar**
To verify, paste this into irb:

ENV.each_key { |k| } == ENV
ENV.each_value { |v| } == ENV
ENV.delete_if { |k, v| false } == ENV
ENV.keep_if { |k, v| true } == ENV
ENV.update({}) == ENV
ENV.update({}) { |name, old, new | new } == ENV
ENV.merge!({}) == ENV
ENV.merge!({}) { |name, old, new | new } == ENV

### CODE REVIEW

File: hash.c [hunk 1]
```diff
@@ -5844,8 +5844,8 @@ env_f_to_hash(VALUE _)
 
 /*
  * call-seq:
- *   ENV.to_h                        -> hash
- *   ENV.to_h {|name, value| block } -> hash
+ *   ENV.to_h                        -> Hash
+ *   ENV.to_h {|name, value| block } -> Hash
```
Comment (nobu) [inline hash.c]: Why `Hash`?
It looks like that the `Hash` class itself will be returned.

Comment (BurdetteLamar) [reply hash.c]: I count 13 "-> Hash" vs. 3 "-> hash", so thought to make consistent.  But I probably should not have mixed that in with the definite doc errors.

Comment (BurdetteLamar) [reply hash.c]: Do you want all Hash, all hash, or leave unchanged?

**Comment by BurdetteLamar** [comment]
Can this be merged?

**Comment by BurdetteLamar** [comment]
Can this be merged?

**Pull request merged**